iothub_client/inc/iothub_client_properties.h

```diff
@@ -0,0 +1,298 @@
+    /** @brief   Version of the structure.  Currently must be IOTHUB_CLIENT_DESERIALIZED_PROPERTY_VERSION_1. */
+    int structVersion;
+    /** @brief   Whether this is a property the device reported (and we're seeing what IoT Hub last received from 
+         the device) or whether it's a writable property request from the device. */
+    IOTHUB_CLIENT_PROPERTY_TYPE propertyType;
+    /** @brief Name of the component.  Optional; may be NULL for the root component. */
+    const char* componentName;
+    /** @brief Name of the property. */
+    const char* name;
+    /** @brief Whether the value is passed as a string or non-null terminated length. */
+    /** @description  Applications can safely assume that this will always be IOTHUB_CLIENT_PROPERTY_VALUE_STRING.  
+    *                 They do not need to write code to handle IOTHUB_CLIENT_PROPERTY_VALUE_BINARY.
+    *                 IOTHUB_CLIENT_PROPERTY_VALUE_BINARY may be supported by the SDK in the future, but 
+    *                 IOTHUB_CLIENT_PROPERTY_VALUE_STRING will *always* by the default.
+    */
+    IOTHUB_CLIENT_PROPERTY_VALUE_TYPE valueType;
+    /** @brief Value of the property. Currently this is only set to str. */
+    union {
+        const char* str;
+        unsigned const char* bin;
+    } value;
+    /** @brief Number of bytes in valueLength.  Does not include null-terminator if IOTHUB_CLIENT_PROPERTY_VALUE_STRING 
+    *          is set */
+    size_t valueLength;
+} IOTHUB_CLIENT_DESERIALIZED_PROPERTY;
+
+/**
+* @brief   Serializes reported properties into a format for sending to IoT Hub.
+*
+* @param[in]   properties                  Pointer to IOTHUB_CLIENT_REPORTED_PROPERTY to be serialized.
+* @param[in]   numProperties               Number of elements contained in @p properties.
+* @param[in]   componentName               Optional component name these properties are part of.  May be NULL for default 
+*                                          component.
+* @param[out]  serializedProperties        Serialized output of @p properties for sending to IoT Hub.
+*                                          The application must release this memory using 
+*                                          IoTHubClient_Serialize_Properties_Destroy.
+*                                          Note: This is NOT a null-terminated string.
+* @param[out]  serializedPropertiesLength  Length of serializedProperties.
+*
+* @remarks  Applications can also manually construct the payload based on the convention rules.  This API is an easier 
+*           to use alternative, so the application can use the more convenient IOTHUB_CLIENT_REPORTED_PROPERTY 
+*           structure instead of raw serialization.
+* 
+*           This API does not perform any network I/O.  It only translates IOTHUB_CLIENT_REPORTED_PROPERTY into a byte 
+*           array.  Applications should use APIs such as IoTHubDeviceClient_LL_SendPropertiesAsync to send the 
```

Comment:
nit: when you reference a function which is documented by doxygen, you can tell doxygen to link to it by putting empty parenthesis after the name:
```suggestion
*           array.  Applications should use APIs such as IoTHubDeviceClient_LL_SendPropertiesAsync() to send the 
```